tests/BookReader.test.js

```diff
@@ -1,59 +1,154 @@
+    expect(BookReader.util.cssPercentage(2, 1)).toEqual(200 + "%");
+  });
 
-test('calculate a percentage suitable for CSS', () => {
-    expect(BookReader.util.cssPercentage(2,1)).toEqual(200+'%');
-});
+  test("notInArray function which return false if value present", () => {
+    expect(BookReader.util.notInArray(4, [4, "Pete", 8, "Shubham"])).toEqual(false);
+    expect(BookReader.util.notInArray(10, ["8", "9", "10", 10 + ""])).toEqual(true);
+  });
 
-test('escapeHTML function which replaces the string', () => {
-    expect(BookReader.util.escapeHTML('Me & You')).toEqual('Me &amp; You');
-    expect(BookReader.util.escapeHTML('Me > You')).toEqual('Me &gt; You');
-    expect(BookReader.util.escapeHTML('Me < You')).toEqual('Me &lt; You');
-    expect(BookReader.util.escapeHTML('Me " You')).toEqual('Me &quot; You');
-});
+  test("escapeHTML function which replaces the string", () => {
+    expect(BookReader.util.escapeHTML("Me & You")).toEqual("Me &amp; You");
+    expect(BookReader.util.escapeHTML("Me > You")).toEqual("Me &gt; You");
+    expect(BookReader.util.escapeHTML("Me < You")).toEqual("Me &lt; You");
+    expect(BookReader.util.escapeHTML('Me " You')).toEqual("Me &quot; You");
+  });
 
-test('Decodes a URI component and converts + to emptyStr', () => {
+  test("Decodes a URI component and converts + to emptyStr", () => {
     expect(BookReader.util.decodeURIComponentPlus("https%3A%2F%2Farchive.org%2Fskr+")).toEqual("https://archive.org/skr ");
     expect(BookReader.util.decodeURIComponentPlus("%3Fx%3D+test")).toEqual("?x= test");
-});
+  });
 
-test('Encodes a URI component and converts emptyStr to +', () => {
+  test("Encodes a URI component and converts emptyStr to +", () => {
     expect(BookReader.util.encodeURIComponentPlus("?x=test ")).toEqual("%3Fx%3Dtest+");
     expect(BookReader.util.encodeURIComponentPlus("ABC abc 123")).toEqual("ABC+abc+123");
+  });
 });
 
-let clock;
+describe("Testing debounce function", () => {
+  let clock;
+
+  beforeEach(() => {
+    clock = sinon.useFakeTimers();
+  });
 
-beforeEach(() => {
-  clock = sinon.useFakeTimers();
+  afterEach(() => {
+    clock.restore();
+  });
+
+  test("testing debounce", () => {
+    const func = jest.fn();
+    const debouncedFunc = BookReader.util.debounce(func, 1000);
+    // Call it immediately
+    debouncedFunc();
+    expect(func).toHaveBeenCalledTimes(0); // func not called
+
+    // Call it several times with 500ms between each call
+    for (let i = 0; i < 10; i++) {
+      clock.tick(500);
+      debouncedFunc();
+    }
+    expect(func).toHaveBeenCalledTimes(0); // func not called
+
+    // wait 1000ms
+    clock.tick(1000);
+    expect(func).toHaveBeenCalledTimes(1); // func called
+  });
 });
 
-afterEach(() => {
-  clock.restore();
+describe('Testing Throttle function', () => {
+  let clock;
+
+  beforeEach(() => {
+    clock = sinon.useFakeTimers();
+  });
+  afterEach(() => {
+    clock.restore();
+  });
```

Comment:
You don't really need the `beforeEach` and `afterEach` if you only have a single test in here. These methods are most useful if you have a bunch of tests that all need the same setup.

Reply:
Okay